### PR TITLE
validation unit tests

### PR DESCRIPTION
**Pull request by pweil-**
Let's get some tests in there :+1:

### CODE REVIEW

File: pkg/validate/validate_test.go [hunk 1]
```diff
@@ -74,63 +74,224 @@ properties:
...
+			expectedErrs: []error{errors.New("orchestratorProfile cannot be nil")},
+		},
+		"orchestrator version good": {
+			f: func(oc *api.OpenShiftManagedCluster) { oc.Properties.OrchestratorProfile.OrchestratorVersion = "v3.10" },
+		},
+		"orchestrator version bad": {
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.OrchestratorProfile.OrchestratorVersion = "" },
+			expectedErrs: []error{errors.New("invalid properties.openShiftVersion \"\"")},
+		},
+		"openshfit config nil": {
```
Comment (jim-minter) [inline pkg/validate/validate_test.go]: spelling

File: pkg/validate/validate.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func validateAgentPoolProfiles(apps []*api.AgentPoolProfile) (errs []error) {
 		appmap[app.Name] = app
 
 		if i > 0 && app.VnetSubnetID != apps[i-1].VnetSubnetID {
-			errs = append(errs, fmt.Errorf("duplicate properties.agentPoolProfiles.vnetSubnetID %q", app.VnetSubnetID))
+			errs = append(errs, fmt.Errorf("invalid properties.agentPoolProfiles.vnetSubnetID %q all subnets must match when using vnetSubnetID", app.VnetSubnetID))
```
Comment (jim-minter) [inline pkg/validate/validate.go]: add colon? `%q: all`

File: pkg/validate/validate_test.go [hunk 3]
```diff
@@ -74,63 +74,224 @@ properties:
...
 		},
 		"test external only false - invalid fqdn fails": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.FQDN = "()" },
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.FQDN = "()" },
 			expectedErrs: []error{errors.New("invalid properties.fqdn \"()\"")},
 			externalOnly: false,
 		},
+		"provisioning state bad": {
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.ProvisioningState = "bad" },
+			expectedErrs: []error{errors.New("invalid properties.provisioningState \"bad\"")},
```
Comment (jim-minter) [inline pkg/validate/validate_test.go]: I wish we used backquotes throughout this file, e.g. errors.New(\`invalid properties.provisioningState "bad"\`)

File: pkg/validate/validate_test.go (hunk 1 above)
Comment (pweil-) [reply pkg/validate/validate_test.go]: fixed

File: pkg/validate/validate.go (hunk 2 above)
Comment (pweil-) [reply pkg/validate/validate.go]: done

File: pkg/validate/validate_test.go (hunk 3 above)
Comment (pweil-) [reply pkg/validate/validate_test.go]: wish granted

**Pull request merged**